### ISSUE TITLE
Reflection: alias inference does not work in macro block

### ISSUE THREAD

**Issue by andreasabel** [agda/agda#1943]
AIM XXIII, Ulf's reflection talk:

``` agda
open import Agda.Builtin.Reflection
open import Agda.Builtin.Unit

-- WORKS
macro
  idT : Term → Term → TC ⊤
  idT = unify

-- FAILS
macro
  test = unify
```

Error:

```
Term → TC ⊤ !=< TC ⊤ of type Set
when checking that the expression unify has type Term → TC ⊤
```

**Comment by andreasabel** [comment]
The offending code is in `Decl.hs`.  It does not postpone upon a meta-variable in the type of the macro.

``` haskell
  -- check macro type if necessary
  when (Info.defMacro i == MacroDef) $ do
    t' <- normalise t
    TelV tel tr <- telView t'
```

telView does not produce the right thing if type t is blocked by a meta.

``` haskell
    let telList = telToList tel
        resType = abstract (telFromList (drop (length telList - 1) telList)) tr
    expectedType <- el primAgdaTerm --> el (primAgdaTCM <#> primLevelZero <@> primUnit)
    equalType resType expectedType
      `catchError` \ _ -> typeError . GenericDocError =<< sep [ text "Result type of a macro must be"
                                                              , nest 2 $ prettyTCM expectedType ]
```

Relevant patches include dbe8cb4d @phile314 and 3e8d104a @UlfNorell.
